apollo-router/src/services/router_service.rs

```diff
@@ -221,7 +221,7 @@ where
 
             Ok(RouterResponse::builder()
                 .errors(errors)
-                .status_code(StatusCode::INTERNAL_SERVER_ERROR)
+                .status_code(StatusCode::BAD_REQUEST)
```

Comment:
IIRC there are some query planner errors that could be a 500? Like if the JS query planner does not answer? Can we discriminate the errors here?

Reply:
Good point ! Fixed